Fix `uninlined_format_args` in compiler crates with the diagnostic migration completed

Noratrieb opened the pull request:
Convert all the crates that have had their diagnostic migration completed (except save_analysis because that will be deleted soon and apfloat because of the licensing problem).

Some of them have been reviewed by myself and they were all correct (though I still recommend going over all of them again for review).

Nilstrieb commented on compiler/rustc_codegen_llvm/src/llvm_util.rs:
```diff
@@ -327,11 +327,11 @@ fn print_target_features(sess: &Session, tm: &llvm::TargetMachine) {
     println!("Features supported by rustc for this target:");
     for (feature, desc) in &rustc_target_features {
-        println!("    {1:0$} - {2}.", max_feature_len, feature, desc);
+        println!("    {feature:max_feature_len$} - {desc}.");
```
That's a fun one, you rarely see it. Looks correct though.

Nilstrieb commented:
I looked through all the changes and they look fine, as expected.

One problem with this is that this will probably cause a bunch of conflicts with diagnostic migrations which is pretty bad. I'm not sure what the best way to proceed is here. Maybe it makes sense to only do the change on crates that have their diagnostics fully migrated?

Noratrieb committed:
Fix `uninlined_format_args` for some compiler crates

Convert all the crates that have had their diagnostic migration
completed (except save_analysis because that will be deleted soon and
apfloat because of the licensing problem).

Nilstrieb commented:
As a user for rust-analyzer, the problems aren't bad enough to not do this now in my opinion, but I could also understand it if we deem it bad enough and want to delay it.

bors commented:
:umbrella: The latest upstream changes (presumably #105880) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

RalfJung commented:
I don't understand why the bot pinged me, I see no interpreter changes?

Nilstrieb commented:
I had some at first that I then removed.

jackh726 commented:
r=me after rebase 

@bors delegate+

bors commented:
:v: @Nilstrieb can now approve this pull request
<!-- homu: {"type":"Delegated","delegator":"jackh726","delegate":"Nilstrieb"} -->

Nilstrieb commented:
@bors r=jackh726

Btw, I have bors permissions already, no need to delegate :D

bors commented:
:pushpin: Commit fd7a15971069156f1741fb3a3c34c9f0c54d5744 has been approved by `jackh726`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=jackh726 fd7a15971069156f1741fb3a3c34c9f0c54d5744 -->
<!-- homu: {"type":"Approved","sha":"fd7a15971069156f1741fb3a3c34c9f0c54d5744","approver":"jackh726","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit fd7a15971069156f1741fb3a3c34c9f0c54d5744 with merge 0853d9677ca903cd650d5c4b93b1c55e7995cf01...
<!-- homu: {"type":"BuildStarted","head_sha":"fd7a15971069156f1741fb3a3c34c9f0c54d5744","merge_sha":"0853d9677ca903cd650d5c4b93b1c55e7995cf01"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/3852218303/jobs/6565703824)
Approved by: jackh726
Pushing 0853d9677ca903cd650d5c4b93b1c55e7995cf01 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"jackh726","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/3852218303/jobs/6565703824"},"merge_sha":"0853d9677ca903cd650d5c4b93b1c55e7995cf01"} -->

rust-timer commented:
Finished benchmarking commit (0853d9677ca903cd650d5c4b93b1c55e7995cf01): [comparison URL](https://perf.rust-lang.org/compare.html?start=b85f57d652a141b5c73f4f46b986a92b6992e9d9&end=0853d9677ca903cd650d5c4b93b1c55e7995cf01&stat=instructions:u).

### Overall result: ✅ improvements - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=b85f57d652a141b5c73f4f46b986a92b6992e9d9&end=0853d9677ca903cd650d5c4b93b1c55e7995cf01&stat=instructions:u)
This is a highly reliable metric that was used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | -     | -              | 0     |
| Improvements ✅ <br /> (primary)   | -     | -              | 0     |
| Improvements ✅ <br /> (secondary) | -0.3% | [-0.3%, -0.3%] | 1     |
| All ❌✅ (primary)                 | -     | -              | 0     |

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=b85f57d652a141b5c73f4f46b986a92b6992e9d9&end=0853d9677ca903cd650d5c4b93b1c55e7995cf01&stat=max-rss)
This benchmark run did not return any relevant results for this metric.

### [Cycles](https://perf.rust-lang.org/compare.html?start=b85f57d652a141b5c73f4f46b986a92b6992e9d9&end=0853d9677ca903cd650d5c4b93b1c55e7995cf01&stat=cycles:u)
<details>
<summary>Results</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean | range        | count |
|:----------------------------------:|:----:|:------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -    | -            | 0     |
| Regressions ❌ <br /> (secondary)  | 2.0% | [2.0%, 2.0%] | 1     |
| Improvements ✅ <br /> (primary)   | -    | -            | 0     |
| Improvements ✅ <br /> (secondary) | -    | -            | 0     |
| All ❌✅ (primary)                 | -    | -            | 0     |
</details>
